lib/services/local/users.go

```diff
@@ -1491,6 +1490,21 @@ func (s *IdentityService) GetKeyAttestationData(ctx context.Context, publicKey c
 	return &resp, nil
 }
 
+func attestedKeyFingerprint(pub crypto.PublicKey) (string, error) {
+	sshPub, err := ssh.NewPublicKey(pub)
+	if err != nil {
+		return "", trace.Wrap(err)
+	}
+
+	key := ssh.FingerprintSHA256(sshPub)
+
+	// The SHA256 hash may contain "//", which is a forbidden pattern for database keys.
+	// We replace this pattern with "slashslash" so that we can insert the key without losing data.
+	key = strings.ReplaceAll(key, "//", "slashslash")
```

Comment:
Shouldn't we base64 it or use URL encoding (something that has standard) instead of replacing it with a random string?

Reply:
+1 for using a proper encoding. `base64.URLEncoding` should do the job.

Reply:
Thanks for this suggestion, I was honestly not sure what the straightforward solution was, base64 encoding of course makes sense.

Reply:
It looks like URL base64 encoding also has invalid characters (`=`). What do you guys think of using a UUID hashed from the public key data? Any other ideas?

Reply:
Would RawURLEncoding work? I think it does `[a-zA-Z]-_` (no padding char).

https://pkg.go.dev/encoding/base64#RawURLEncoding